ragna/source_storages/_chroma.py

```diff
@@ -100,22 +101,19 @@ def retrieve(
                 max(int(num_tokens * 2 / chunk_size), 100),
                 collection.count(),
             ),
-            include=["distances", "metadatas", "documents"],
+            include=include,
         )
 
         num_results = len(result["ids"][0])
-        result = {
-            key: [None] * num_results if value is None else value[0]  # type: ignore[index]
-            for key, value in result.items()
-        }
+        result = {key: result[key][0] for key in ["ids", *include]}
         # dict of lists -> list of dicts
         results = [
-            {key[:-1]: value[idx] for key, value in result.items()}
+            {key: value[idx] for key, value in result.items()}
             for idx in range(num_results)
         ]
 
         # That should be the default, but let's make extra sure here
-        results = sorted(results, key=lambda r: r["distance"])
+        results = sorted(results, key=lambda r: r["distances"])
```

Comment:
Question: was this a bug?

Reply:
Sorry, I should have elaborated on this. I previously thought that it would be a good idea to turn the plural forms into singulars when going from dict of lists to list of dicts, i.e.

```python
{"distances": [...], "ids": [...]}
```

to

```
[{"distance": ..., "id": ...}, ...]
```

From a language standpoint this makes sense given that each result in the new list now has a singular distance, ID, etc. However, this makes it harder to map the Chroma documentation to the results that we want to return and I think we should value this higher than correct language. Thus, I've removed the transforming of the plural into the singular form by cutting of the final charactly (L113, `key[:-1]` -> L111, `key`).